pallets/amm/src/lib.rs

```diff
@@ -156,6 +159,38 @@ pub mod pallet {
                 let pool_liquidity_amount: PoolLiquidityAmount =
                     Self::pools(base_asset, quote_asset);
 
+                let optimal_quote_amount = Self::quote(
+                    base_amount,
+                    pool_liquidity_amount.base_amount,
+                    pool_liquidity_amount.quote_amount,
+                );
+
+                let (ideal_base_amount, ideal_quote_amount): (Balance, Balance) =
+                    if optimal_quote_amount <= quote_amount {
+                        (base_amount, optimal_quote_amount)
+                    } else {
+                        let optimal_base_amount = Self::quote(
+                            quote_amount,
+                            pool_liquidity_amount.base_amount,
+                            pool_liquidity_amount.quote_amount,
+                        );
+                        (optimal_base_amount, quote_amount)
+                    };
+
+                let (minimum_base_amount, minimum_quote_amount) = match is_inverted {
+                    true => (minimum_amounts.1, minimum_amounts.0),
+                    false => (minimum_amounts.0, minimum_amounts.1),
+                };
```

Comment:
I think this can actually be rewritten to this
```suggestion
                let (minimum_base_amount, minimum_quote_amount) = if is_inverted {
                    (minimum_amounts.1, minimum_amounts.0)
                } else {
                    (minimum_amounts.0, minimum_amounts.1)
                };
```